### PR TITLE
Allow for neg. ULs in CountsStatistics

### PR DESCRIPTION
**Pull request by adonath**
**Description**

This pull request changes `compute_upper_limits` top allow for negative upper limits. While physically this doesn't make sense, statistically it is a correct solution and points to underestimated systematics. This change is also done for consistency with the `TSMapEstimator` I noticed the difference in https://github.com/adonath/notebooks-public/blob/master/excess-vs-ts-map-estimator.ipynb.

**Dear reviewer**

### CODE REVIEW

File: docs/estimators/index.rst [hunk 1]
```diff
@@ -104,6 +104,20 @@ n_sigma_ul        Number of sigma used for upper limit estimation
 ts_threshold_ul   TS threshold to define the use of an upper limit
 ================= =================================================
 
+A note on negative flux and upper limit values:
+
+.. note::
+
+    Gammapy allows for negative flux values and upper limits by default.
```
Comment (registerrier) [inline docs/estimators/index.rst:111]: Technically, this depends on the allowed parameter range, no? 
If you force, `amplitude` and  `norm` to be positive quantities the UL will be positive as well right?

**Review by registerrier — DISMISSED** [review]
Thanks @adonath .

My main comment is regarding is the explanation text which might not be fully true since by defining parameter range the allowed upper bound of the interval will change.

File: docs/estimators/index.rst (hunk 1 above)
Comment (adonath) [reply docs/estimators/index.rst:111]: Indeed I forgot about the recent change, that allows taking into account the boundaries. I'll adapt the text accordingly.

**Review by registerrier — APPROVED** [review]
Thanks @adonath !

**Pull request merged**